monitor/src/main.rs

```diff
@@ -0,0 +1,167 @@
+mod config;
+mod controller;
+mod external_api;
+mod event_listener;
+
+use crate::config::{Config, SingleConfig};
+use crate::event_listener::{get_github_repo_url, listen_to_commits, listen_to_pull_requests};
+use crate::external_api::{add_configuration, delete_configuration, get_actions_file, get_configuration_by_id, get_configurations, update_configuration, AppState};
+use actix_web::{web, App, HttpServer};
+use clap::{Arg, Command};
+use controller::send_to_controller;
+use tokio::task::JoinHandle;
+use std::path::Path;
+use std::sync::Arc;
+use tokio;
+use tokio::sync::RwLock;
+
+#[tokio::main]
+async fn main() -> std::io::Result<()> {
+    let config = get_config();
+    let configs = Arc::new(RwLock::new(config));
+
+    let mut thread_listeners_handles: Vec<JoinHandle<()>> = vec![];
+
+    println!("Launching API and listening to events on Github repository...");
+
+    tokio::select! {
+        _ = launch_api_server(Arc::clone(&configs)) => {},
+        _ = launch_github_listeners(Arc::clone(&configs), &mut thread_listeners_handles) => {},
+    }
+
+    println!("Threads launched, waiting for them to finish...");
+    println!("{:#?}", thread_listeners_handles);
+
+    futures::future::join_all(thread_listeners_handles).await;
+
+    Ok(())
+}
+
+fn get_config() -> Config {
+    // CLI arguments
+    let matches = Command::new("GitHub Monitor")
+        .version("1.0")
+        .about("Monitors a GitHub repository for changes")
+        .arg(Arg::new("config")
+            .short('c')
+            .long("config")
+            .required(false)
+            .help("The path to the config file"))
+        .arg(Arg::new("event")
+            .short('e')
+            .long("event")
+            .required(false)
+            .help("The event to listen to (commit, pull_request, *)"))
+        .arg(Arg::new("repo_owner")
```

Comment:
Same thing that for the repo_name

Reply:
same answer, we would rather keep it like that.